design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+| 12 | File conflict between two secret groups | You are using SP in "file" mode and have defined multiple secret groups | One or more secret groups have he same ouput file. | Error: You have provided conflicting file paths; secret groups "{value}", "{value}", ... all have output file "{value}".<br><br>SP fails to start | | |
+
+#### Integration / E2E tests
+
+#### Security testing
+
+Security testing will include:
+
+- Automated Vulnerability scans
+
+####  Performance testing
+
+As described in the [Performance](#performance) section above, the performance
+of the Secrets Provider for Push-to-File mode will be manually measured using
+the following metrics:
+
+- Latency (processing time) for parsing Pod annotations
+- Latency for generating secrets requests and processing responses
+- Latency for writing retrieved secrets to files
+
+_**QUESTION: Do we need to measure CPU and memory footprints?**_
+
+The measurements will be made using a representative configuration in terms
+of the number of secrets groups and the number of secrets per secrets group.
+Details are TBD, but a representative configuration might look something like
+this:
+
+- 5 secrets groups
+- 5 secrets / secrets group
+- Mix of secret value lengths (10 to 20 characters, plus some base64-encoded
+  TLS certificates)
+
+## Logs
+
+| **Scenario** | **Log Level** | **Notes** |
+| ------------ | ------------- | --------- |
+| Annotation has `conjur.org` prefix, but invalid key or value format | `Error` | Note 1 |
+| Annotation has `conjur.org` prefix, but unknown annotation type | `Info` | Note 2 |
+| Start of secrets retrieval for a group | `Info` | Includes group name |
+| Start of secrets file creation for a group | `Info` | Includes group name |
+| Error processing Go template | `Error` | Note 3 |
+| Error writing secrets file | `Error` | Note 3 |
+
+- *Note 1:* Annotation values are treated as non-sensitive and will be included the logs.
+- *Note 2:* For forward compatability, annotation keys with the prefix
+  `conjur.org` but an unexpected annotation type will be logged at `Info` level.
+- *Note 3:* Care must be taken to prevent secrets file content from leaking.
+
+## Documentation
+
+Documentation for the M1 Secrets Provider Push to File feature will include:
+
+- Update the
+  [Secrets Provider init container configuration documentation](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Integrations/k8s-ocp/cjr-k8s-secrets-provider-ic.htm).
+- Update the [Secrets Provider application container configuration documentation](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Integrations/k8s-ocp/cjr-k8s-secrets-provider-ac.htm).
```

Comment:
(to document using annotations for config, right?)

should we have an error flow for mixing env var and annotation config?